### PR TITLE
#786: Update sequence diagram for Bulk Prepare and GET Bulk Transfer

### PR DESCRIPTION
**Pull request by oderayi**
- Updated sequence diagram for bulk prepare to reflect code changes
- Added page and sequence diagram for `GET /bulkTransfer/{ID}` endpoint
- Fixed a minor issue with bulk abort sequence diagram
- Bumped version

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -100,6 +100,7 @@
       * [Timeout Overview](mojaloop-technical-overview/central-bulk-transfers/transfers/3.1.0-transfer-timeout-overview-for-bulk.md)
         * [Timeout Handler Consume](mojaloop-technical-overview/central-bulk-transfers/transfers/3.1.1-transfer-timeout-handler-consume.md)  
       * [Bulk Abort Overview](mojaloop-technical-overview/central-bulk-transfers/transfers/4.1.0-transfer-abort-overview-for-bulk.md)
+      * [Bulk Lookup Overview](mojaloop-technical-overview/central-bulk-transfers/transfers/5.1.0-transfer-lookup-overview-for-bulk.md)
```
Comment (elnyry-sam-k) [inline SUMMARY.md:103]: @oderayi - "lookup" is used for ALS / Discovery phase so we need to refactor this terminology... Please use "get-bulk-transfer" or something to that effect

File: mojaloop-technical-overview/central-bulk-transfers/transfers/5.1.0-transfer-lookup-overview-for-bulk.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+## Bulk Transfer Lookup Overview
+
+Sequence design diagram for the Bulk Transfer Lookup process.
+
+## Sequence Diagram
+
+{% uml src="mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-lookup-overview.plantuml" %}
```
Comment (elnyry-sam-k) [inline mojaloop-technical-overview/central-bulk-transfers/transfers/5.1.0-transfer-lookup-overview-for-bulk.md:7]: please refactor "lookup" as discussed above

File: mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        ilpPacket
+    end hnote
+    TRANSFER_DAO <-- DB: Return query result
+    deactivate DB
+    GET_HANDLER <-- TRANSFER_DAO: Return transfer status
+    deactivate TRANSFER_DAO
+    
+    alt Is there a bulk transfer with the given ID recorded in the system?
+        alt Request is from Payee FSP?
+            GET_HANDLER <-> GET_HANDLER: Filter out transfers not included in original request to Payee FSP, if any\n (e.g. in partial fulfilment scenario).
```
Comment (rmothilal) [inline mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml:97]: here I would state the status that should be filtered out. i.e ABORTED_REJECTED, EXPIRED_RESERVED, COMMITTED

File: mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+            end note
+            NOTIFY_HANDLER <- GET_HANDLER: Publish notification event
+            DFSP <- NOTIFY_HANDLER: callback PUT on /bulkTransfers/{ID}
+        else transferState for individual transfers in [ABORTED_REJECTED, EXPIRED_RESERVED]
+            note left of GET_HANDLER #yellow
+                {
+                    "bulkTransferState": <bulk-state>,
+                    "individualTransferResults": [
+                        {
+                            "transferState": <state>,
```
Comment (rmothilal) [inline mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml:139]: please revisit the api spec and look at the IndividualTransferResult objectand align your individualTransferResults with that

File: mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        alt Request is from Payee FSP?
+            GET_HANDLER <-> GET_HANDLER: Filter out transfers not included in original request to Payee FSP, if any\n (e.g. in partial fulfilment scenario).
+        end
+        alt Yes AND transferState for all indvidual transfers is COMMITTED \nThis implies that a succesful transfer with the given ID is recorded in the system
+            note left of GET_HANDLER #yellow
+                {
+                    "bulkTransferState": "COMMITTED",
+                    "individualTransferResults": [
+                        {
+                            "fulfilment": "WLctttbu2HvTsa1XWvUoGRcQozHsqeu9Ahl2JW9Bsu8",
```
Comment (rmothilal) [inline mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml:105]: please revisit the api spec and look at the IndividualTransferResult objectand align your individualTransferResults with that

File: mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+end box
+box "Central Ledger" #LightYellow
+    participant TOPIC_GET
+    participant GET_HANDLER
+    participant TRANSFER_DAO
+    participant DB
+end box
+
+' start flow
+group Request transfer status
```
Comment (elnyry-sam-k) [inline mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml:55]:
```suggestion
group Request Bulk transfer status
```

File: mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+	participant NOTIFY_HANDLER
+end box
+box "Central Ledger" #LightYellow
+    participant TOPIC_GET
+    participant GET_HANDLER
+    participant TRANSFER_DAO
+    participant DB
+end box
+
+' start flow
```
Comment (elnyry-sam-k) [inline mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml:54]: @oderayi I think we need to perform the validation (whether the request was from either the Payer or the Payee FSP) first even before requesting the bulk transfer status here, I think. Why would we need it if the validation fails?

File: mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+        DFSP <- NOTIFY_HANDLER: callback PUT on /bulkTransfers/{ID}
+        deactivate NOTIFY_HANDLER
+        note right of NOTIFY_HANDLER #lightgray
+            Log ERROR event
+        end note
+    else A bulk transfer with the given ID is not present in the System or this is an invalid request
+        note left of NOTIFY_HANDLER #yellow
+            {
+                "errorInformation": {
+                    "errorCode": <integer>,
```
Comment (elnyry-sam-k) [inline mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml:165]:
```suggestion
                    "errorCode": 3210,
```

File: mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+                    "errorDescription": "Client error description"
```
Comment (elnyry-sam-k) [inline mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml:166]:
```suggestion
                    "errorDescription": "Bulk transfer ID not found"
```

**Comment by elnyry-sam-k** [comment]
Looks good Steven, a few changes needed for which I made specific comments. Please review

File: mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml (hunk 9 above)
Comment (oderayi) [reply mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-5.1.0-get-overview.plantuml:166]: Thanks @elnyry-sam-k

File: mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-4.1.0-abort-overview.plantuml [hunk 10]
```diff
@@ -124,6 +124,21 @@ group DFSP2 sends a Fulfil Abort Transfer request
     end note
     DFSP2 ->> BULKAPI: **PUT - /bulkTransfers/<ID>/error**
     activate BULKAPI
+    
+    BULKAPI -> CACHE: Persist request payload with messageId in cache
```
Comment (rmothilal) [inline mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-4.1.0-abort-overview.plantuml:128]: i wouldn't call this cache, its an object store called mojaloop-object-store (**MLOS**) (refer to https://github.com/oderayi/documentation/blob/feature/%23786-bulk-transfers-diagrams/mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-1.1.0-bulk-prepare-overview.plantuml) , please ensure all diagrams that use mongodb are updated to this

Comment (oderayi) [reply mojaloop-technical-overview/central-bulk-transfers/assets/diagrams/sequence/seq-bulk-4.1.0-abort-overview.plantuml]: Agreed! Thanks @rmothilal

**Pull request merged**